app/src/main/kotlin/com/proxerme/app/helper/MaterialDrawerHelper.kt

```diff
@@ -35,22 +36,42 @@ class MaterialDrawerHelper(context: Activity, toolbar: Toolbar,
 
     private val header: AccountHeader
     private val drawer: Drawer
+    //Only available if the device is a tablet
+    private val miniDrawer: MiniDrawer?
+    private val crossfader: Crossfader<*>?
 
+    private var stickyItemIds : Array<DrawerItem> = arrayOf()
     private var currentItem: DrawerItem? = null
 
     init {
         header = buildAccountHeader(context, savedInstanceState)
         drawer = buildDrawer(context, toolbar, header, savedInstanceState)
+        if (ScreenHelper.isTablet(context)) {
+            miniDrawer = drawer.miniDrawer.withIncludeSecondaryDrawerItems(true)
+            crossfader = buildCrossfader(context, drawer, miniDrawer, savedInstanceState)
+            miniDrawer.withCrossFader(CrossfadeWrapper(crossfader!!))
+            crossfader.getCrossFadeSlidingPaneLayout().setShadowResourceLeft(R.drawable.material_drawer_shadow_left)
+        } else {
+            miniDrawer = null
+            crossfader = null
+        }
+
         currentItem = DrawerItem.fromOrNull(savedInstanceState?.getLong(STATE_CURRENT_DRAWER_ITEM_ID))
     }
 
     fun onBackPressed(): Boolean {
-        if (isDrawerOpen()) {
+        if(crossfader?.isCrossFaded() ?: false) {
+            crossfader?.crossFade()
+
+            return true
+        } else if (isDrawerOpen()) {
             drawer.closeDrawer()
 
             return true
         } else {
-            val startPage = PreferenceHelper.getStartPage(drawer.drawerLayout.context)
+            //DrawerLayout is only null if the device is a tablet (and has a crossfader)
+            val startPage = PreferenceHelper.getStartPage(drawer.drawerLayout?.context
+                    ?: crossfader!!.getCrossFadeSlidingPaneLayout()!!.context)
```

Comment:
Hmm, is there a better way than this? I don't like nonnull assertions, since you lose safety.